streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -1227,24 +1227,30 @@ object ZSink extends ZSinkPlatformSpecificConstructors with Serializable {
   /**
    * Creates a sink accumulating incoming values into a map.
    * Key of each element is determined by supplied function.
+   * Combines elements with same key with supplied function f.
+   *
    */
-  final def collectAllToMap[K, A](key: A => K): ZSink[Any, Nothing, Nothing, A, Map[K, A]] =
-    foldLeft[A, Map[K, A]](Map.empty[K, A])((map, element) => map + (key(element) -> element))
+  final def collectAllToMap[K, A](key: A => K)(f: (A, A) => A): ZSink[Any, Nothing, Nothing, A, Map[K, A]] =
```

Comment:
Because this and `collectAllToMapN` are now implemented in terms of `foldLeft`, which is a Sink that has no leftover, please change the leftover of these two sinks from `A` to `Nothing` to better document them.

Reply:
Good catch. thanks @vasilmkd